ee/elf-loader/src/loader/loader.c

```diff
@@ -57,17 +62,33 @@ static void wipeUserMem(void)
 //End of func:  void wipeUserMem(void)
 //--------------------------------------------------------------
 // *** MAIN ***
+// 
 //--------------------------------------------------------------
 int main(int argc, char *argv[])
 {
 	static t_ExecData elfdata;
-	int ret;
+	int ret, i;
+
+	elfdata.epc = 0;
 
 	GS_BGCOLOUR = WHITE_BG;
-	if (argc < 1) {  // arg1=path to ELF
+	// arg[0]=path to ELF
+	// arg[1] partition if exists otherwise is ""
+	if (argc < 2) {  
 		GS_BGCOLOUR = RED_BG;
 		return -EINVAL;
 	}
+
+	char *new_argv[argc - 1];
+	int fullPath_length = 1 + strlen(argv[1]) + strlen(argv[0]);
+	char fullPath[fullPath_length];
+	strcpy(fullPath, argv[1]);
+	strcat(fullPath, argv[0]);
```

Comment:
can you add +1 to len and tack a ':' on to the end of argv[1]? or is it better to just include it in partition name when calling?

Reply:
Have in mind that the concatenation of parameters, always happen, so when you load an `elf` that doesn't contain partition it is just `"" + mass:/your.elf`, however, if I add the `:` then it will break the path of the parameter.

I think that is "developer responsibility" to call the function putting in the partition parameter, something finished with the `:`

Reply:
Oh ok fair enough, didn’t think of that